### PR TITLE
Build all target families in a CI job

### PR DESCRIPTION
**Pull request by multiplemonomials**
### Summary of changes 

I feel like we have had lots of issues where someone (often me) changes a file for target X and that causes a build failure for target Y.  I think we should put an end to that -- things like this are basically why CI was invented!  Will this cause us to get rate limited by Github Actions?  No idea!  But let's find out!

#### Impact of changes 

Submitting a pull request will now cause Mbed to be compiled for every major target family.

### Documentation 

----------------------------------------------------------------------------------------------------------------
### Pull request type 

    [] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [X] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results 

    [] No Tests required for this change (E.g docs only update)
    [] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    [X] This is a test-only change
    
----------------------------------------------------------------------------------------------------------------

### CODE REVIEW

File: connectivity/drivers/emac/TARGET_RENESAS_EMAC/TARGET_RZ_A1XX/CMakeLists.txt [hunk 1]
```diff
@@ -9,5 +9,5 @@ target_include_directories(mbed-emac
 target_sources(mbed-emac
     PRIVATE
         rza1_emac.cpp
-        rza1_eth.cpp
+        rza1_eth.c
```
Comment (multiplemonomials) [inline connectivity/drivers/emac/TARGET_RENESAS_EMAC/TARGET_RZ_A1XX/CMakeLists.txt:12]: This has been broken for a long time in Mbed CLI 2... guess no one ever tried to build for this target O_O

File: targets/TARGET_Maxim/TARGET_MAX32670/Libraries/PeriphDrivers/Source/SYS/sys_me15.c [hunk 2]
```diff
@@ -44,7 +44,6 @@
 #include "mxc_assert.h"
 #include "mxc_sys.h"
 #include "mxc_delay.h"
-#include "aes.h"
```
Comment (multiplemonomials) [inline targets/TARGET_Maxim/TARGET_MAX32670/Libraries/PeriphDrivers/Source/SYS/sys_me15.c:47]: This header does not seem to exist for MAX32670, and deleting the include doesn't seem to cause any compile issues for this .c file

File: targets/TARGET_STM/stm_dma_utils.c [hunk 3]
```diff
@@ -396,6 +396,15 @@ IRQn_Type stm_get_dma_irqn(const DMALinkInfo *dmaLink)
                 case 4:
                 case 5:
                     return DMA1_Ch4_7_DMA2_Ch1_5_DMAMUX1_OVR_IRQn;
+#elif defined(TARGET_MCU_STM32F0)
```
Comment (multiplemonomials) [inline targets/TARGET_STM/stm_dma_utils.c:399]: DMA code was missing some logic for this specific STM32F0 chip

File: targets/TARGET_Maxim/TARGET_MAX32670/CMakeLists.txt [hunk 4]
```diff
@@ -2,6 +2,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 add_subdirectory(TARGET_MAX32670EVKIT EXCLUDE_FROM_ALL)
+add_subdirectory(TARGET_XDOT_MAX32670 EXCLUDE_FROM_ALL)
```
Comment (multiplemonomials) [inline targets/TARGET_Maxim/TARGET_MAX32670/CMakeLists.txt:5]: Fixing missing add_subdirectory for this target

File: connectivity/drivers/emac/TARGET_STM/stm32xx_emac.cpp [hunk 5]
```diff
@@ -306,13 +306,13 @@ bool STM32_EMAC::low_level_init_successful()
     tr_info("PHY ID %#X", PHY_ID);
 
     /* Initialize Tx Descriptors list: Chain Mode */
-    if (HAL_ETH_DMATxDescListInit(&EthHandle, DMATxDscrTab, &Tx_Buff[0][0], ETH_TXBUFNB) != HAL_OK) {
+    if (HAL_ETH_DMATxDescListInit(&EthHandle, DMATxDscrTab, reinterpret_cast<uint8_t *>(&Tx_Buff[0][0]), ETH_TXBUFNB) != HAL_OK) {
```
Comment (multiplemonomials) [inline connectivity/drivers/emac/TARGET_STM/stm32xx_emac.cpp:309]: Oops... guess I broke this in my ethernet refactor PR...

**Review by JohnK1987 — APPROVED** [review]

**Pull request merged**